src/features/change-language/ui/ChangeLanguageSelector.tsx

```diff
@@ -0,0 +1,52 @@
+import { FC } from 'react';
+
+import { useNavigation } from '@react-navigation/native';
+import { useTranslation } from 'react-i18next';
+
+import { colors } from '@shared/lib';
+import {
+  YStack,
+  XStack,
+  Box,
+  BoxProps,
+  Text,
+  ArrowRightIcon,
+} from '@shared/ui';
+
+const ChangeLanguageSelector: FC<BoxProps> = props => {
+  const { t, i18n } = useTranslation();
+  const navigation = useNavigation();
+  const languagesAvailable = Object.keys(i18n.services.resourceStore.data);
+  const { resolvedLanguage } = i18n;
+
+  const changeLanguage = async (locale: string) => {
+    await i18n.changeLanguage(locale);
+    navigation.goBack();
+  };
+
+  return (
+    <Box {...props}>
```

Comment:
This is not necessary to rewrite. Just a note.
If you have a `YStack`/`XStack`/`ZStack` as root element in a component then it is not necessary to additionally wrap it in `Box`, because `YStack`/`XStack`/`ZStack` are `Box`es under the hood and accept `BoxProps`.